android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/INIParser.java

```diff
@@ -0,0 +1,123 @@
+package com.tavultesoft.kmea.util;
+
+import java.util.Map;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * A custom, lightweight INI parser designed initially for use with Keyman's .kmp and .inf files.
+ * @author jahorton
+ *
+ */
+public class INIParser {
+	public static Map<String, Map<String, List<String>>> parse(String _string) throws INIParseException {
+		Map<String, Map<String, List<String>>> sections = new HashMap<String, Map<String, List<String>>>();
+		
+		String[] linesArray = _string.split("\n");
```

Comment:
Will this cope with `\r\n` from Windows?

Reply:
Split will automatically leave a `\r` on the end of each line.  Fortunately, that's technically a whitespace character, meaning `trim()` will nix it.